zerver/lib/bugdown/__init__.py

```diff
@@ -1001,6 +1001,36 @@ def run(self, root: Element) -> None:
                 else:
                     add_embed(root, url, extracted_data)
 
+class CleanUpProcessor(markdown.treeprocessors.Treeprocessor):
+    """Applies final logic related to user mentions after all the other processors"""
+
+    def handle_quoted_user_mentions(self, root: Element) -> Element:
+        if not self.markdown.zulip_message:
+            return root
+
+        # Just keep the HTML for styling purposes for mentions inside blockquotes.
+        for quote in walk_tree(root, lambda e: e if e.tag == "blockquote" else None):
+            for mention_tag in walk_tree(quote, lambda e: e if e.get('class') == "user-mention" else None):
+                current_id = mention_tag.get('data-user-id')
+                if current_id:
+                    mention_tag.set('data-silent-user-id', current_id)
```

Comment:
Do we need to set data-silent-user-id field for anything?

Reply:
No, I've added this instead of changing all the different places that handle notifications and narrowing to mentions, etc. on the frontend.

It might break some apps (ZT?) that might be depending on each rendered `user-mention` to have a `data-user-id`.